.github/workflows/ci-test.yml

```diff
@@ -42,10 +42,22 @@ jobs:
         chmod +x bazel_installer.sh
         sudo ./bazel_installer.sh
 
+    - name: Cache pip
+      uses: actions/cache@v2
+      with:
+        # This path is specific to Ubuntu
+        path: ~/.cache/pip
+        # Look to see if there is a cache hit for the corresponding setup.py + TFX version 
+        key: ${{ runner.os }}-pip-${{ hashFiles('tfx/dependencies.py') }}-
+        restore-keys: |
+          ${{ runner.os }}-pip-${{ hashFiles('tfx/dependencies.py') }}-
+          ${{ runner.os }}-pip-
```

Comment:
Out of curiosity. Why do we need this second restore-key?

And it seems like we don't need to repeat the one defined in "key" in restore-key. (https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#using-the-cache-action)

Reply:
Second key is used to restore from cache a partial hit (for example if changes in deps files, it will retrieve the previous cache). This allows to reuse at least the shared deps wheels even if dependencies have changed.

Reply:
(also removed the first key)

Reply:
Thank you for the explanation! I should have pay more attention on the documentation.... It seems great!